### DISCUSSION TITLE
Webhooks on user registration

### DISCUSSION THREAD

**Discussion by x610s** [zitadel/zitadel#8968] [Q&A] [answered]
is there a way to call a webhook on user registration? 

Im making this onboarding: https://zitadel.com/docs/guides/integrate/onboarding/end-users

**Comment by hifabienne** [comment]
Our new implementation of actions can be added to api calls, in this case the managmenet api is triggered with the create human user reqeust. you could add an action, which is basically a webhook there: https://zitadel.com/docs/apis/resources/action_service_v3

**Reply by codersaadi** [reply]
what about when using the repository "zitadel/typescript" locally . 
should we add a more direct logic on user registration there ?
or use zitadel actions ?
i am talking about "webhooks" with oss zitadel version

**Reply by hifabienne** [reply]
This would both be possible. I think it depends on your preference

**Reply by codersaadi** [reply]
> This would both be possible. I think it depends on your preference


Thanks but
yeah i know both are possible . 
v1 actions were very scoped and not ideal , i am glad we have v2, but **the question is are they stable enough to be used in a b2b2c application**? or enough only for user management. 

i mean we could have our own solutions as we will be using oss version i guess (depends , tech decision). 

i hope some sensible discussion will also benefit others and clears a lot of misconception regarding data synchronization with zitadel. 
what would be more ideal and realistic , even if we directly call the backend http-edge endpoint from 'zitadel/typescript' that would be webhook. 
i agree v1 should be discouraged but 
should v2 be encouraged?

**Reply by hifabienne** [reply, marked as answer]
At the moment they are still marked as beta, goal would be to move them to a stable state with one of the next releases.
I would recommend testing your use case with the actions v2 and see if everything works fine for you. In that case i would encourage you to use the actions v2 already.

**Reply by codersaadi** [reply]
> At the moment they are still marked as beta, goal would be to move them to a stable state with one of the next releases. I would recommend testing your use case with the actions v2 and see if everything works fine for you. In that case i would encourage you to use the actions v2 already.

Thanks, i am glad that i found the answer even in beta state v2 structure is very practical and looks promising too .i was thinking about building with both approaches (switchable with env variables) .i guess  for user management, and data synchronization they will work just fine as mention in the docs.
